index.bs

```diff
@@ -543,54 +544,85 @@ To <dfn>handle an incoming message</dfn> given a [=WebSocket connection=]
     Issue: Should we instead close |connection| with [=status
     codes|status code=] 1003, or [=respond with an error=]?
 
- 2. [=Assert=]: |data| is a [=scalar value string=], because the
+ 1. [=Assert=]: |data| is a [=scalar value string=], because the
      WebSocket [=handling errors in UTF-8-encoded data=] would already
      have [=fail the WebSocket connection|failed the WebSocket
      connection=] otherwise.
 
     Issue: Nothing seems to define what [=status codes|status code=]
     is used for UTF-8 errors.
 
- 3. Let |parsed| be the result of [=parse JSON into Infra
-    values|parsing JSON into Infra values=] given |data|. If this
-    throws an exception, then [=respond with an error=] given
-    |connection| and [=error code=] [=invalid argument=], and finally
-    return.
-    <!-- corresponds to Parse error (-32700) in JSON-RPC -->
-
- 5. Let |result| be the result of [=process a command|processing a
+ 1. Let |result| be the result of [=process a command|processing a
     command=] given |data|.
 
- 6. If |result| is an [=Error=], then [=respond with an error=] given
-    |connection|, |result|, and |parsed|["<code>id</code>"], and finally
-    return.
+ 1. If |result| is an [=error=], then [=respond with an error=] given
+    |connection|, |data|, and |result|'s [=error code=], and finally return.
+
+ 1. Assert: |result| is a [=success=]
 
- 7. Let |response| be a new [=map=] with the following properties:
+ 1. Let |command id| and |value| be the two fields |result|'s data.
```

Comment:
Alright, so this is unpacking the return value from 'Return success with (data parsed["id"], value)' elsewhere.